api/webhook.go

```diff
@@ -335,31 +324,20 @@ func PostWebhook(c *gin.Context) {
 				return
 			}
 		}
+	}
 
-		// files is empty if the build event is pull_request
-		if len(files) == 0 {
-			number, err := parsePRNumberFromRef(b.GetRef())
-			if err != nil {
-				retErr := fmt.Errorf("%s: failed to get pull_request number for %s: %v", baseErr, r.GetFullName(), err)
-				util.HandleError(c, http.StatusInternalServerError, retErr)
-
-				h.SetStatus(constants.StatusFailure)
-				h.SetError(retErr.Error())
-
-				return
-			}
-
-			// send API call to capture list of files changed for the pull request
-			files, err = source.FromContext(c).ChangesetPR(u, r, number)
-			if err != nil {
-				retErr := fmt.Errorf("%s: failed to get changeset for %s: %v", baseErr, r.GetFullName(), err)
-				util.HandleError(c, http.StatusInternalServerError, retErr)
+	// check if the build event is a pull_request
+	if strings.EqualFold(b.GetEvent(), constants.EventPull) && webhook.PRNumber != 0 {
```

Comment:
This was the main culprit. Previously, at this point, if we were dealing with a push event that had no changeset (which was processed above) we would end up here with `len(files)` being zero. We would then continue processing the request as if it was a pull request causing `500` errors while trying to extract pull request info that doesn't exist.